src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -0,0 +1,412 @@
+package mesosphere.mesos
+
+import mesosphere.marathon.MarathonConf
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.pod.PodDefinition
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.raml
+import mesosphere.marathon.state.{ EnvVarString, PathId, Timestamp }
+import mesosphere.mesos.ResourceMatcher.ResourceSelector
+import org.apache.mesos.{ Protos => mesos }
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+
+object TaskGroupBuilder {
+  val log = LoggerFactory.getLogger(getClass)
+
+  def build(
+    podDefinition: PodDefinition,
+    offer: mesos.Offer,
+    runningTasks: => Iterable[Task],
+    newInstanceId: PathId => Instance.Id,
+    config: MarathonConf
```

Comment:
I want to avoid the pattern used in `TaskBuilder` here, where there's a class with a single (public) function that gets called like this: `val result = new TaskBuilder(...).build(...)`. Having that function in an object is IMO the better way here. OTOH it would make sense to create a _private_ case class in this build function that makes it easier for the other functions here to access some variables without having to pass them in the function signature.